File: src/main/java/org/customer/controller/AccountController.java
package org.customer.controller;



import org.customer.dto.AccountDTO;
import org.customer.service.AccountService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;


@RestController
@RequestMapping("/accounts")
class AccountController {
  @Autowired
  private AccountService accountService;

  @PostMapping("/open")
  public AccountDTO createAccount(@RequestParam String customerId, @RequestParam double initialCredit) {
    return accountService.createAccount(customerId, initialCredit);
  }
}

File: src/main/java.kotlin/org/customer/controller/AccountController.kt
package org.customer.controller

import org.customer.dto.AccountDTO
import org.customer.service.AccountService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping("/accounts")
internal class AccountController {
    @Autowired
    private val accountService: AccountService? = null

    @PostMapping("/open")
    fun createAccount(@RequestParam customerId: String?, @RequestParam initialCredit: Double): AccountDTO? {
        return accountService.createAccount(customerId, initialCredit)
    }
}
